src/Language/DifferentialDatalog/Validate.hs

```diff
@@ -414,8 +411,13 @@ exprValidate1 _ _ ctx (EPHolder p)        = do
                    CtxStruct EStruct{..} _ f -> "Missing field " ++ f ++ " in constructor " ++ exprConstructor
                    _               -> "_ is not allowed in this context"
     check (ctxPHolderAllowed ctx) p msg
-exprValidate1 d _ ctx (EBinding p v _)    = checkNoVar p d ctx v
+exprValidate1 d _ ctx (EBinding p v _)    = do
+    checkNoVar p d ctx v
+    
 exprValidate1 d tvs _ (ETyped _ _ t)      = typeValidate d tvs t
+exprValidate1 _ _ ctx (ERef p _)          =
+    -- Rust does not allow pattern matching inside 'Arc'
```

Comment:
this is a strange comment. what should matter is what ddlog allows, not what rust allows.

Reply:
Sure, but it is really hard to implement things that don't directly map to Rust. The comment just documents the reason for the limitation.